integration/hello_command_test.go

```diff
@@ -0,0 +1,179 @@
+	assert.Equal(t, must.NotFail(actual.Get("maxBsonObjectSize")), int32(16777216))
+	assert.Equal(t, must.NotFail(actual.Get("maxMessageSizeBytes")), int32(48000000))
+	assert.Equal(t, must.NotFail(actual.Get("maxMessageSizeBytes")), int32(48000000))
+	assert.Equal(t, must.NotFail(actual.Get("maxWriteBatchSize")), int32(100000))
+	assert.IsType(t, must.NotFail(actual.Get("localTime")), time.Time{})
+	assert.IsType(t, must.NotFail(actual.Get("connectionId")), int32(1))
+	assert.Equal(t, must.NotFail(actual.Get("minWireVersion")), int32(0))
+	assert.Equal(t, must.NotFail(actual.Get("maxWireVersion")), int32(21))
+	assert.Equal(t, must.NotFail(actual.Get("readOnly")), false)
+	assert.Equal(t, must.NotFail(actual.Get("ok")), float64(1))
+}
+
+func TestHelloWithSupportedMechs(t *testing.T) {
+	t.Parallel()
+
+	ctx, collection := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
+	db := collection.Database()
+
+	usersPayload := []bson.D{
+		{
+			{"createUser", "hello_user"},
+			{"roles", bson.A{}},
+			{"pwd", "hello_password"},
+		},
+		{
+			{"createUser", "hello_user_scram1"},
+			{"roles", bson.A{}},
+			{"pwd", "hello_password"},
+			{"mechanisms", bson.A{"SCRAM-SHA-1"}},
+		},
+		{
+			{"createUser", "hello_user_scram256"},
+			{"roles", bson.A{}},
+			{"pwd", "hello_password"},
+			{"mechanisms", bson.A{"SCRAM-SHA-256"}},
+		},
+	}
+
+	if !setup.IsMongoDB(t) {
+		usersPayload = append(usersPayload, primitive.D{
+			{"createUser", "hello_user_plain"},
+			{"roles", bson.A{}},
+			{"pwd", "hello_password"},
+			{"mechanisms", bson.A{"PLAIN"}},
+		})
+	}
+
+	for _, u := range usersPayload {
+		require.NoError(t, db.RunCommand(ctx, u).Err())
+	}
+
+	testCases := map[string]struct { //nolint:vet // used for test only
+		user  string
+		mechs *types.Array
+		err   string
+	}{
+		"NotFound": {
+			user: db.Name() + ".not_found",
+		},
+		"AnotherDB": {
+			user: db.Name() + "_not_found.another_db",
+		},
+		"HelloUser": {
+			user:  db.Name() + ".hello_user",
+			mechs: must.NotFail(types.NewArray("SCRAM-SHA-1", "SCRAM-SHA-256")),
+		},
+		"HelloUserPlain": {
+			user:  db.Name() + ".hello_user_plain",
+			mechs: must.NotFail(types.NewArray("PLAIN")),
+		},
+		"HelloUserSCRAM1": {
+			user:  db.Name() + ".hello_user_scram1",
+			mechs: must.NotFail(types.NewArray("SCRAM-SHA-1")),
+		},
+		"HelloUserSCRAM256": {
+			user:  db.Name() + ".hello_user_scram256",
+			mechs: must.NotFail(types.NewArray("SCRAM-SHA-256")),
+		},
+		"EmptyUsername": {
+			user:  db.Name() + ".",
+			mechs: nil,
+		},
+		"MissingSeparator": {
+			user: db.Name(),
+			err:  "UserName must contain a '.' separated database.user pair",
+		},
+	}
+
+	for name, tc := range testCases {
+		tc, name := tc, name
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var res bson.D
+
+			if tc.mechs != nil && tc.mechs.Contains("PLAIN") {
+				setup.SkipForMongoDB(t, "PLAIN authentication mechanism is not support by MongoDB")
```

Comment:
1. Should we use `Deprecated:` in the doc so that gopls tells us to use the newer method?
2. Do we really want to do it? I ask because there are cases where we might deviate from what MongoDB does (for example, we support PLAIN) and having an expected test failure for MongoDB in such case doesn't seem to be very valuable & adds pressure to the test suite, in my opinion.

Reply:
Example of something that can't be properly tested if we replace SkipForMongoDB with FailsForMongoDB: `TestHelloWithSupportedMechs/HelloUserPlain` because we require something (user account with PLAIN method) which cannot be created with MongoDB. What do you think about leaving SkipForMongoDB as it was before due to situations like this?

Reply:
"Just try it and push changes" to all points.

I don't understand the second comment. Are you saying that the helper that was specifically created to test expected test failures can't be used in a test that would fail? Anyway, just show the problem in code with actual code.

Reply:
I think second point would fail, as soon as we attempt to create user with `PLAIN` for MongoDB. Let me take a look at this.

Reply:
> I think second point would fail, as soon as we attempt to create user with PLAIN for MongoDB

If it would fail on `t` returned by `FailsForMongoDB`, that would be an expected failure, isn't it?

Reply:
My sentence was confusing, I was agreeing with the fact that `FailsForMongoDB` returns expected value :sweat:. So expectation can be met. I didn't mean _fail_ as test would fail

Reply:
Maybe we should start using "xfail" :)

Reply:
That might have been wiser choice of the vocab :hugs: